sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd_functions.md

```diff
@@ -0,0 +1,647 @@
+                        simd_mask<N / VS> mask, PropertyListT props = {});
+/*acc-2*/ void prefetch(AccessorT acc, simd<OffsetT, N / VS> byte_offsets,
+                        PropertyListT props = {});
+
+// The next 2 are similar (acc-1,2). `VS` parameter is set to 1,
+// which allows callers to omit explicit specification of `T` and `N` params.
+template <typename T, int N, typename AccessorT, typename OffsetT,
+          typename PropertyListT = empty_properties_t>
+/*acc-3*/ void prefetch(AccessorT acc, simd<OffsetT, N> byte_offsets,
+                        simd_mask<N> mask, PropertyListT props = {});
+/*acc-4*/ void prefetch(AccessorT acc, simd<OffsetT, N> byte_offsets,
+                        PropertyListT props = {});
+
+// The next 2 are similar to (acc-1,2), added to handle `byte_offsets` in `simd_view` form.
+template <typename T, int N, int VS = 1, typename OffsetSimdViewT,
+          typename AccessorT, typename PropertyListT = empty_properties_t>
+/*acc-5*/ void prefetch(AccessorT acc, OffsetSimdViewT byte_offsets,
+                        simd_mask<N / VS> mask, PropertyListT props = {});
+/*acc-6*/ void prefetch(AccessorT acc, OffsetSimdViewT byte_offsets,
+                        PropertyListT props = {});
+
+/// The next functions perform transposed 1-channel prefetch.
+template <typename T, int VS = 1, typename AccessorT, typename OffsetT,
+          typename PropertyListT = empty_properties_t>
+/*acc-7*/ void prefetch(AccessorT acc, OffsetT byte_offset, simd_mask<1> mask,
+                        PropertyListT props = {});
+/*acc-8*/ void prefetch(AccessorT acc, OffsetT byte_offset,
+                        PropertyListT props = {});
+template <typename T, int VS = 1, typename AccessorT,
+          typename PropertyListT = empty_properties_t>
+/*acc-9*/ void prefetch(AccessorT acc, simd_mask<1> mask, PropertyListT props = {});
+/*acc-10*/ void prefetch(AccessorT acc, PropertyListT props = {});
+```
+### Description
+`(usm-1,2,3,4,5,6)`: Prefetches the memory locations addressed by the base USM pointer `ptr` or the accessor `acc` and the vector of 64-bit byte-offsets `byte_offsets`.
+
+`(acc-1,2,3,4,5,6)`: Prefetches the memory locations addressed by the base USM pointer `ptr` or the accessor `acc` and the vector of byte-offsets `byte_offsets`.
+The `byte_offsets` is a vector of 32-bit integers elements in [stateful](#statelessstateful-memory-mode) mode, and it is a vector of 64-bit integer elements for in [stateless](#statelessstateful-memory-mode) mode.
+
+`(usm-7,8,9,10)`: Prefetches a linear block of memory addressed by the base USM pointer `ptr` or the accessor `acc` and the optional 64-bit `byte-offset`.
+
+`(acc-7,8,9,10)`: Prefetches a linear block of memory addressed by the base USM pointer `ptr` or the accessor `acc` and the optional `byte-offset` parameter, which is 32-bit in [stateful](#statelessstateful-memory-mode) mode, and 32-bit in [stateless](#statelessstateful-memory-mode) mode.
```

Comment:
"and 32-bit in stateless" is that correct for prefetch?

Reply:
Fixed in 161e753c6100f9bd74ec32fde32237c6957aa4c9 and 0be57b6470981fb5078dbb17241f04bd6e3c8c55